public/app/features/plugins/AppRootPage.tsx

```diff
@@ -90,26 +101,56 @@ class AppRootPage extends Component<Props, State> {
     this.setState({ nav });
   };
 
+  renderRoute = (route: RouteDescriptor) => {
+    const { plugin } = this.state;
+    const roles = route.roles ? route.roles() : [];
+
+    if (!plugin) {
+      return;
+    }
+
+    return (
+      <Route
+        exact={!route.soft}
+        path={`/a/${plugin.meta.id}${route.path}`}
```

Comment:
Nice! I like how apps will be able to manage their own sub-routes.

Reply:
Can app plugins just put in a Router that is scoped to `/a/${plugin.meta.id}`  what am I missing?  why do we want so much control?

Reply:
There is no such a thing in react-router word as scoped, nested router @ryantxu

Reply:
I'm slightly skeptical about doing it now - updating to RR v6-beta which supports nested routes. I'll peek at how much of an effort that is for us to upgrade. Then we could follow the convention @jdbranham uses in his app.